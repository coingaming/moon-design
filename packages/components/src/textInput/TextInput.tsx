import * as React from 'react';
import styled from 'styled-components';

import { Label } from '../private/label/label';

import { Input } from './Input';
import { InputError } from './Error';

/**
 * Types & Settings
 */
type TextInputTypes =
  | 'date'
  | 'datetime-local'
  | 'email'
  | 'number'
  | 'password'
  | 'search'
  | 'tel'
  | 'text'
  | 'url'
  | 'time'
  | 'url';

type TextInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  label?: string;
  type?: TextInputTypes;
  placeholder?: string;
  error?: boolean | string;
  rounded?: boolean;
};

const TextInputElem = styled(Input as any)(({ error, theme: { color } }) => ({
  '&:focus': {
    borderColor: !error ? color.piccolo[100] : color.chiChi[100],
    outline: 'none',
  },
}));

/**
 * Component
 *
 * 1. Leaving the placeholder as an empty string by default allows us to float
 *    the label when a user starts typing, even if a placeholder isn't defined.
 */
const TextInput: React.FC<TextInputProps> = ({
  type = 'text',
  disabled,
  placeholder = ' ',
  label,
  error,
  rounded,
  ...props
}) => {
  const inputProps = {
    disabled,
    type,
    placeholder,
    rounded,
    ...props,
  };

  const TextInputInner = () => (
    <TextInputElem rounded={!!rounded} error={!!error} {...inputProps} />
  );

  if (!label) {
    return TextInputInner();
  }

  return (
    <Label text={label} disabled={disabled}>
      <>
        {TextInputInner()}
        {error && <InputError>{error}</InputError>}
      </>
    </Label>
  );
};

export { TextInputProps };

export default TextInput;
