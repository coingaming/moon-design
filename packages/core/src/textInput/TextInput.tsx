import React, { forwardRef } from 'react';
import { ColorProps } from '@heathmont/moon-themes';
import Size from '../private/enums/Size';
import TextInputBasic from './private/TextInputBasic';
import TextInputInnerLabel from './private/TextInputInnerLabel';
import TextInputPasswort from './private/TextInputPasswort';
import TextInputSizeType from './private/types/SizeTypes';
import TextInputTypes from './private/types/TextInputTypes';

export interface TextInputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  id?: string;
  inputSize: TextInputSizeType | string;
  type: TextInputTypes | string;
  label?: JSX.Element | string;
  placeholder?: string;
  hintText?: JSX.Element | string;
  isError?: boolean;
  dir?: 'ltr' | 'rtl' | 'auto';
  showPasswordText?: JSX.Element | string;
  backgroundColor?: ColorProps;
  isSharpLeftSide?: boolean;
  isSharpRightSide?: boolean;
  isSharpTopSide?: boolean;
  isSharpBottomSide?: boolean;
  isTopBottomBorderHidden?: boolean;
  isSideBorderHidden?: boolean;
}

const TextInput = forwardRef<HTMLInputElement, TextInputProps>((props, ref) => {
  const { inputSize, type } = props;

  // render input with show/hide password
  if (type === 'password') {
    return <TextInputPasswort {...props} ref={ref} />;
  }
  if (inputSize === Size.XLARGE) {
    return <TextInputInnerLabel {...props} ref={ref} />;
  }
  // render rest sizes
  return <TextInputBasic {...props} ref={ref} />;
});

export default TextInput;
