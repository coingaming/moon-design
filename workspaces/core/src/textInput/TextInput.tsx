import React, { forwardRef } from 'react';
import TextInputBasic from './private/TextInputBasic';
import TextInputInnerLabel from './private/TextInputInnerLabel';
import TextInputPassword from './private/TextInputPassword';
import type TextInputProps from './private/types/TextInputProps';

const TextInput = forwardRef<HTMLInputElement, TextInputProps>((props, ref) => {
  const { inputSize = 'md', type } = props;

  //render input with show/hide password
  if (type === 'password') {
    return <TextInputPassword {...props} ref={ref} />;
  }
  if (inputSize === 'xl') {
    return <TextInputInnerLabel {...props} ref={ref} />;
  }
  // render rest sizes
  return <TextInputBasic {...props} ref={ref} />;
});

export default TextInput;
