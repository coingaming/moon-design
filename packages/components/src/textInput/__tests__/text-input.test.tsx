import React from 'react';
import { sportsbetDark, ThemeProvider } from '@heathmont/moon-themes';
import { create } from 'react-test-renderer';
import 'jest-styled-components';

import TextInput from '../TextInput';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={sportsbetDark}>{component}</ThemeProvider>
);

describe('TextInput', () => {
  test('renders correctly with `text` type by default', () => {
    const textInput = create(
      renderWithTheme(<TextInput label="Example Input" />)
    );

    expect(textInput).toMatchSnapshot();
  });

  test('renders correctly without a label', () => {
    const textInput = create(
      renderWithTheme(<TextInput placeholder="Enter code" />)
    );

    expect(textInput).toMatchSnapshot();
  });

  test('renders correctly for other types', () => {
    const textInput = create(
      renderWithTheme(
        <fieldset>
          <TextInput label="Text" type="text" required />
          <TextInput label="Date" type="date" required />
          <TextInput label="Email" type="email" required />
          <TextInput label="Number" type="number" required />
          <TextInput label="Password" type="password" required />
          <TextInput label="Search" type="search" required />
          <TextInput label="Telephone" type="tel" required />
          <TextInput label="URL" type="url" required />
        </fieldset>
      )
    );
    expect(textInput).toMatchSnapshot();
  });

  test('renders as `disabled` when defined', () => {
    const textInput = create(
      renderWithTheme(
        <fieldset>
          <TextInput label="Text" type="text" disabled />
          <TextInput label="Date" type="date" disabled />
          <TextInput label="Email" type="email" disabled />
          <TextInput label="Number" type="number" disabled />
          <TextInput label="Password" type="password" disabled />
          <TextInput label="Search" type="search" disabled />
          <TextInput label="Telephone" type="tel" disabled />
          <TextInput label="URL" type="url" disabled />
        </fieldset>
      )
    );
    expect(textInput).toMatchSnapshot();
  });

  test('renders as an `error` when defined', () => {
    const textInput = create(
      renderWithTheme(
        <fieldset>
          <TextInput label="Text" type="text" error />
          <TextInput label="Date" type="date" error />
          <TextInput label="Email" type="email" error />
          <TextInput label="Number" type="number" error />
          <TextInput label="Password" type="password" error />
          <TextInput label="Search" type="search" error />
          <TextInput label="Telephone" type="tel" error />
          <TextInput label="URL" type="url" error />
        </fieldset>
      )
    );
    expect(textInput).toMatchSnapshot();
  });

  test('renders with rounded prop', () => {
    const textInput = create(
      renderWithTheme(<TextInput rounded label="Text" type="text" error />)
    );
    expect(textInput).toMatchSnapshot();
  });
});
