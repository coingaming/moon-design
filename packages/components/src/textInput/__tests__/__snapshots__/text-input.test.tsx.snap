// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders as \`disabled\` when defined 1`] = `
.c0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #8697A2;
  font-size: 1rem;
  opacity: 0.5;
}

.c1 {
  width: 100%;
  max-width: 100%;
  padding: 0.5rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: #1A212A;
  border: 1px solid #31373F;
  border-radius: 0.75rem;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:hover:not(:focus):not([disabled]) {
  border-color: #4C5159;
}

.c1:not(:placeholder-shown):not([type="date"]):invalid {
  border-color: #FF4E64;
}

.c1:invalid,
.c1:-moz-ui-invalid {
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1:focus {
  border-color: #0CCF61;
  outline: none;
}

<fieldset>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      Text
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="text"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      Date
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="date"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      Email
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="email"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      Number
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="number"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      Password
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="password"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      Search
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="search"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      Telephone
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="tel"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
      disabled={true}
    >
      URL
    </span>
    <input
      className="c1"
      disabled={true}
      placeholder=" "
      type="url"
    />
  </label>
</fieldset>
`;

exports[`TextInput renders as an \`error\` when defined 1`] = `
.c0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #8697A2;
  font-size: 1rem;
}

.c2 {
  display: inline-block;
  margin-top: 0.25rem;
  font-size: 0.75rem;
  color: #FF4E64;
}

.c1 {
  width: 100%;
  max-width: 100%;
  padding: 0.5rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: #1A212A;
  border: 1px solid #31373F;
  border-radius: 0.75rem;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
  border-color: #FF4E64;
}

.c1::-webkit-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:hover:not(:focus):not([disabled]) {
  border-color: #4C5159;
}

.c1:not(:placeholder-shown):not([type="date"]):invalid {
  border-color: #FF4E64;
}

.c1:invalid,
.c1:-moz-ui-invalid {
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1:focus {
  border-color: #FF4E64;
  outline: none;
}

<fieldset>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Text
    </span>
    <input
      className="c1"
      placeholder=" "
      type="text"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Date
    </span>
    <input
      className="c1"
      placeholder=" "
      type="date"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Email
    </span>
    <input
      className="c1"
      placeholder=" "
      type="email"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Number
    </span>
    <input
      className="c1"
      placeholder=" "
      type="number"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Password
    </span>
    <input
      className="c1"
      placeholder=" "
      type="password"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Search
    </span>
    <input
      className="c1"
      placeholder=" "
      type="search"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Telephone
    </span>
    <input
      className="c1"
      placeholder=" "
      type="tel"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      URL
    </span>
    <input
      className="c1"
      placeholder=" "
      type="url"
    />
    <p
      className="c2"
      role="alert"
    />
  </label>
</fieldset>
`;

exports[`TextInput renders correctly for other types 1`] = `
.c0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #8697A2;
  font-size: 1rem;
}

.c1 {
  width: 100%;
  max-width: 100%;
  padding: 0.5rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: #1A212A;
  border: 1px solid #31373F;
  border-radius: 0.75rem;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:hover:not(:focus):not([disabled]) {
  border-color: #4C5159;
}

.c1:not(:placeholder-shown):not([type="date"]):invalid {
  border-color: #FF4E64;
}

.c1:invalid,
.c1:-moz-ui-invalid {
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1:focus {
  border-color: #0CCF61;
  outline: none;
}

<fieldset>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Text
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="text"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Date
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="date"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Email
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="email"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Number
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="number"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Password
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="password"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Search
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="search"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      Telephone
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="tel"
    />
  </label>
  <label
    className=""
  >
    <span
      className="c0"
    >
      URL
    </span>
    <input
      className="c1"
      placeholder=" "
      required={true}
      type="url"
    />
  </label>
</fieldset>
`;

exports[`TextInput renders correctly with \`text\` type by default 1`] = `
.c0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #8697A2;
  font-size: 1rem;
}

.c1 {
  width: 100%;
  max-width: 100%;
  padding: 0.5rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: #1A212A;
  border: 1px solid #31373F;
  border-radius: 0.75rem;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:hover:not(:focus):not([disabled]) {
  border-color: #4C5159;
}

.c1:not(:placeholder-shown):not([type="date"]):invalid {
  border-color: #FF4E64;
}

.c1:invalid,
.c1:-moz-ui-invalid {
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1:focus {
  border-color: #0CCF61;
  outline: none;
}

<label
  className=""
>
  <span
    className="c0"
  >
    Example Input
  </span>
  <input
    className="c1"
    placeholder=" "
    type="text"
  />
</label>
`;

exports[`TextInput renders correctly without a label 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
  padding: 0.5rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: #1A212A;
  border: 1px solid #31373F;
  border-radius: 0.75rem;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
}

.c0::-webkit-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c0::-moz-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c0:-ms-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c0::placeholder {
  color: #8697A2;
  opacity: 1;
}

.c0:hover:not(:focus):not([disabled]) {
  border-color: #4C5159;
}

.c0:not(:placeholder-shown):not([type="date"]):invalid {
  border-color: #FF4E64;
}

.c0:invalid,
.c0:-moz-ui-invalid {
  box-shadow: none;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0:focus {
  border-color: #0CCF61;
  outline: none;
}

<input
  className="c0"
  placeholder="Enter code"
  type="text"
/>
`;

exports[`TextInput renders with rounded prop 1`] = `
.c0 {
  display: block;
  margin-bottom: 0.5rem;
  color: #8697A2;
  font-size: 1rem;
}

.c2 {
  display: inline-block;
  margin-top: 0.25rem;
  font-size: 0.75rem;
  color: #FF4E64;
}

.c1 {
  width: 100%;
  max-width: 100%;
  padding: 0.5rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: #FFFFFF;
  background-color: #1A212A;
  border: 1px solid #31373F;
  border-radius: 0.75rem;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
  border-color: #FF4E64;
}

.c1::-webkit-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::-moz-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:-ms-input-placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1::placeholder {
  color: #8697A2;
  opacity: 1;
}

.c1:hover:not(:focus):not([disabled]) {
  border-color: #4C5159;
}

.c1:not(:placeholder-shown):not([type="date"]):invalid {
  border-color: #FF4E64;
}

.c1:invalid,
.c1:-moz-ui-invalid {
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c1:focus {
  border-color: #FF4E64;
  outline: none;
}

<label
  className=""
>
  <span
    className="c0"
  >
    Text
  </span>
  <input
    className="c1"
    placeholder=" "
    type="text"
  />
  <p
    className="c2"
    role="alert"
  />
</label>
`;
